provisioning/miniprov/hare_mp/main.py

```diff
@@ -254,21 +255,202 @@ def test(args):
         exit(-1)
 
 
+def executecmds(cmd: List[str]) -> list:
+    process = subprocess.Popen(cmd,
+                               stdin=subprocess.PIPE,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.PIPE,
+                               encoding='utf8')
+    resp = list(process.communicate())
+    resp.append(str(process.returncode))
+
+    return resp
```

Comment:
The whole method could be looking like this:

```python
    def _execute(cmd: List[str]) -> str:
        process = Popen(cmd,
                        stdin=PIPE,
                        stdout=PIPE,
                        stderr=PIPE,
                        encoding='utf8')
        logging.debug('Issuing CLI command: %s', cmd)
        out, err = process.communicate()
        exit_code = process.returncode
        logging.debug('Finished. Exit code: %d', exit_code)
        if exit_code:
            raise CliException(out, err, exit_code)
        return out
```
So if everything is fine, it will just return the `stdout`. In case of non-zero exit code an exception is thrown (with stderr and exact exit code) - the exception type is custom.

Why do you want to return a list of stdout, stderr and exit code? This result value is not comfortable to handle.

Reply:
used function which is already in use